entrypoint.sh

```
@@ -9,5 +9,12 @@ if [[ "${ENABLE_RAY_SERVE}" == "true" ]]; then
   uv run $ENV_ARG api.py
 else
   echo "🚀 Starting with Uvicorn..."
-  uv run --no-dev $ENV_ARG uvicorn api:app --host 0.0.0.0 --port ${APP_iPORT:-8080} --reload --workers ${API_NUM_WORKERS:-1}
+  # --reload is a development feature (watches/auto-imports source on change)
+  # and is incompatible with multiple workers. Enable it only with
+  # UVICORN_RELOAD=true for local dev; production runs without it.
+  RELOAD_ARG=""
+  if [[ "${UVICORN_RELOAD}" == "true" ]]; then
+    RELOAD_ARG="--reload"
+  fi
+  uv run --no-dev $ENV_ARG uvicorn api:app --host 0.0.0.0 --port ${APP_iPORT:-8080} $RELOAD_ARG --workers ${API_NUM_WORKERS:-1}
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)